src/main/java/com/google/sps/servlets/data/util/GmapsHelper.java

```diff
@@ -0,0 +1,150 @@
+package com.google.sps.data.util;
+
+import com.google.gson.Gson;
+import com.google.maps.GaeRequestHandler;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlacesApi;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceType;
+import com.google.maps.model.PlacesSearchResponse;
+import com.google.maps.model.PlacesSearchResult;
+import com.google.maps.model.RankBy;
+import java.io.IOException;
+import java.util.Arrays;
+
+public class GmapsHelper {
+
+  GeoApiContext context;
+
+  private final PlaceType[] recreationTypes = {
+    PlaceType.AMUSEMENT_PARK,
+    PlaceType.AQUARIUM,
+    PlaceType.ART_GALLERY,
+    PlaceType.BOWLING_ALLEY,
+    PlaceType.CAMPGROUND,
+    PlaceType.GYM,
+    PlaceType.MOVIE_THEATER,
+    PlaceType.MUSEUM,
+    PlaceType.NIGHT_CLUB,
+    PlaceType.PARK,
+    PlaceType.SPA,
+    PlaceType.STADIUM,
+    PlaceType.SHOPPING_MALL,
+    PlaceType.TOURIST_ATTRACTION
+  };
+
+  private final PlaceType[] diningTypes = {
+    PlaceType.BAKERY,
+    PlaceType.BAR,
+    PlaceType.CAFE,
+    PlaceType.BOWLING_ALLEY,
+    PlaceType.MEAL_DELIVERY,
+    PlaceType.MEAL_TAKEAWAY,
+    PlaceType.RESTAURANT
+  };
+
+  private final PlaceType[] groceryTypes = {
+    PlaceType.GROCERY_OR_SUPERMARKET, PlaceType.PHARMACY, PlaceType.PET_STORE, PlaceType.DRUGSTORE
+  };
+
+  public GmapsHelper(boolean useAppEngineContextBuilder) {
+
+    if (useAppEngineContextBuilder) {
+      this.context =
+          new GeoApiContext.Builder(new GaeRequestHandler.Builder())
+              .apiKey("AIzaSyCRG4mbpl4ZOXTmB8xACmqAcnnDSDdLJFU")
+              .build();
+    } else {
+      this.context =
+          new GeoApiContext.Builder().apiKey("AIzaSyCRG4mbpl4ZOXTmB8xACmqAcnnDSDdLJFU").build();
+    }
+  }
+
+  /**
+   * Converts an int Id to a CategoryGroup enum
+   *
+   * @param Id: An integer between 2-4 (inclusive)
+   * @return The corresponding CategoryGroup to Id
+   */
+  public CategoryGroup getCategoryGroupById(int Id) {
+    switch (Id) {
+      case 2:
+        return CategoryGroup.RECREATION;
+      case 3:
+        return CategoryGroup.DINING;
+      case 4:
+        return CategoryGroup.GROCERY;
+      default:
+        return CategoryGroup.GROCERY;
```

Comment:
why default is grocery?

Reply:
Ah yeah I wanted to ask about this but completely forgot. I'm not 100% sure what to here. I put Grocery as the default just because it is the one with the least amount of PlaceTypes, but I think ultimately I want to throw an error here instead, because if the CategoryGroup isn't one of those then it should probably return BAD REQUEST to the client.